src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/SynthesizedStateMachineProperty.cs

```diff
@@ -48,6 +48,11 @@ public override ImmutableArray<CustomModifier> TypeCustomModifiers
             get { return _getter.ReturnTypeCustomModifiers; }
         }
 
+        internal override ushort CountOfCustomModifiersPrecedingByRef
```

Comment:
It feels like we might want to combine TypeCustomModifiers and CountOfCustomModifiersPrecedingByRef into a combined optional entity.  CountOfCustomModifiersPrecedingByRef  makes sense only in the presence of TypeCustomModifiers.

It might be too big of a change right now though.